### ISSUE TITLE
(language name) Short description of issue...

### ISSUE THREAD

**Issue by meatball133** [highlightjs/highlight.js#3730]
**Describe the issue**
Using `//` in crystal will make the highlighter mark the rest of the file.
In crystal is `//` used for integer division.
Code:
```
def something()
    something // 4
end
```
![image](https://user-images.githubusercontent.com/69751659/222831570-0fd37f43-d9b6-41a9-a184-e518381b335a.png)

**Which language seems to have the issue?**
Crystal

**Are you using `highlight` or `highlightAuto`?**
highlight

**Sample Code to Reproduce**
```
def something()
    something // 4
end
```
![image](https://user-images.githubusercontent.com/69751659/222831570-0fd37f43-d9b6-41a9-a184-e518381b335a.png)

**Expected behavior**
The expected behavior is to not mark the rest of the code with the color green.

**Additional context**
Nope

**Comment by joshgoebel** [comment]
The trick here would probably be to introduce a new mode rule that only matches the literal `//` (with no scope, so as not to color it)... we only want to prevent It from being mis-recognized by the regex mode rule.

**Comment by meatball133** [comment]
Yeah, to be a valid regex it needs to be at least 1 character between the // otherwise it is invalid and is thereby integer divison

**Comment by carlqt** [comment]
Hi, I want to contribute but I don't know how to add `crystal` in `tools/developer.html` to test. Anyone can help me out?

**Comment by joshgoebel** [comment]
`./build -t browser crystal`

**Comment by zain-asif-dev** [comment]
I opened a PR to fix this: #4431

The root cause is that the regex-detection heuristic in the Crystal grammar was borrowed from Ruby, where // has no special meaning, so it was treated as an empty regex literal instead of as Crystal's integer division operator. Added a dedicated mode for // and //= that takes priority, and removed the ambiguous empty-regex variant that let this slip through at the start of a line/statement too.
